static/app/views/automations/components/automationBuilderContext.tsx

```diff
@@ -0,0 +1,405 @@
+import {createContext, type Reducer, useCallback, useContext, useReducer} from 'react';
+
+import type FormModel from 'sentry/components/forms/model';
+import {
+  type DataConditionGroup,
+  DataConditionGroupLogicType,
+  type DataConditionType,
+} from 'sentry/types/workflowEngine/dataConditions';
+
+export interface AutomationBuilderState {
+  actionFilters: DataConditionGroup[];
+  triggers: DataConditionGroup;
+}
+
+export interface AutomationActions {
+  addIf: () => void;
+  addIfCondition: (groupIndex: number, conditionType: DataConditionType) => void;
+  addWhenCondition: (conditionType: DataConditionType) => void;
+  removeIf: (groupIndex: number) => void;
+  removeIfCondition: (groupIndex: number, conditionIndex: number) => void;
+  removeWhenCondition: (index: number) => void;
+  updateIfCondition: (
+    groupIndex: number,
+    conditionIndex: number,
+    comparison: Record<string, any>
+  ) => void;
+  updateIfLogicType: (groupIndex: number, logicType: DataConditionGroupLogicType) => void;
+  updateWhenCondition: (index: number, comparison: Record<string, any>) => void;
+  updateWhenLogicType: (logicType: DataConditionGroupLogicType) => void;
```

Comment:
I am slightly concerned that keeping these in sync with the Action types is going to be annoying

Reply:
There might be some TypeScript magic that makes this automatic, but I'd suggest leaving a comment/warning above this at least.

Reply:
is this referring to the state updating actions, or the `Actions` on the `DataConditions`? (i was worried this naming would be confusing 😭)

Reply:
yeahh i'm not sure of a better way to do this, left a comment for now in https://github.com/getsentry/sentry/pull/89233/commits/c3b26dc656daeca63eb34f33b18ab0b48f40ce34